Failed to build with rustc 1.21.0-nightly (215e0b10e 2017-08-08)

Due to: rust-lang/rust@453ad8122c975dc82f6090f0c755d38932ccefc5
Logs:
```
error[E0599]: no method named `for_each_relevant_impl` found for type `&rustc::ty::TraitDef` in the current scope
   --> .cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.149/src/derive.rs:101:23
    |
101 |         peq_trait_def.for_each_relevant_impl(cx.tcx, ty, |impl_id| {
    |                       ^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

error: Could not compile `clippy_lints`.
warning: build failed, waiting for other jobs to finish...
error: failed to compile `clippy v0.0.149`, intermediate artifacts can be found at `/tmp/cargo-install.hM50konMQ6a3`

Caused by:
  build failed
```

--- a/clippy_lints/src/derive.rs
+++ b/clippy_lints/src/derive.rs
@@ -95,10 +95,8 @@ fn check_hash_peq<'a, 'tcx>(
         match_path_old(&trait_ref.path, &paths::HASH),
         let Some(peq_trait_def_id) = cx.tcx.lang_items.eq_trait()
     ], {
-        let peq_trait_def = cx.tcx.trait_def(peq_trait_def_id);
-
         // Look for the PartialEq implementations for `ty`
-        peq_trait_def.for_each_relevant_impl(cx.tcx, ty, |impl_id| {
+        cx.tcx.for_each_relevant_impl(peq_trait_def_id, ty, |impl_id| {
             let peq_is_automatically_derived = is_automatically_derived(&cx.tcx.get_attrs(impl_id));
 
             if peq_is_automatically_derived == hash_is_automatically_derived {

